internal/testtypes/matchers.go

```diff
@@ -0,0 +1,275 @@
+		{"metadata", "uid"},
+		{"metadata", "generation"},
+		{"metadata", "creationTimestamp"},
+		{"metadata", "resourceVersion"},
+		{"metadata", "managedFields"},
+		{"metadata", "deletionGracePeriodSeconds"},
+		{"metadata", "deletionTimestamp"},
+		{"metadata", "ownerReferences"},
+		{"metadata", "selfLink"},
+		{"metadata", "generateName"},
+	}
+)
+
+// Matcher is a Gomega matcher used to establish equality between two Kubernetes runtime.Objects.
+type Matcher struct {
+	// original holds the object that will be used to Match.
+	original runtime.Object
+
+	// diff contains the delta between the two compared objects.
+	diff []byte
+
+	// options holds the options that identify what should and should not be matched.
+	options *MatchOptions
+}
+
+// EqualObject applies the MatchOptions and returns a Matcher for the passed Kubernetes runtime.Object. This function
+// can be used directly as a Gomega Matcher in Gomega Assertions.
+func EqualObject(original runtime.Object, opts ...MatchOption) *Matcher {
+	matchOptions := &MatchOptions{}
+	matchOptions = matchOptions.ApplyOptions(opts)
+
+	// set the allowPaths to '*' by default to not exclude any paths from the comparison.
+	if len(matchOptions.allowPaths) == 0 {
+		matchOptions.allowPaths = [][]string{{"*"}}
+	}
+	return &Matcher{
+		options:  matchOptions,
+		original: original,
+	}
+}
+
+// Match compares the object it's based on to the passed object and returns true if the objects are the same according to
+// the Matcher MatchOptions.
+func (m *Matcher) Match(other interface{}) (success bool, err error) {
+	// Nil checks required first here for:
+	//     1) Nil equality which returns true
+	//     2) One object nil which returns an error
+	otherIsNil := reflect.ValueOf(other).IsNil()
+	originalIsNil := reflect.ValueOf(m.original).IsNil()
+
+	if otherIsNil && originalIsNil {
+		return true, nil
+	}
+	if otherIsNil || originalIsNil {
+		return false, fmt.Errorf("can not compare an object with a nil. original :%v , other %v", m.original, other)
+	}
+
+	// Calculate diff returns a json diff between the two objects.
+	m.diff, err = m.calculateDiff(other)
+	if err != nil {
+		return false, err
+	}
+	return bytes.Equal(m.diff, []byte("{}")), nil
+}
+
+// FailureMessage returns a message comparing the full objects after an unexpected failure to match has occurred.
+func (m *Matcher) FailureMessage(other interface{}) (message string) {
+	return fmt.Sprintf("the following fields were expected to match but did not:\n%s\n%s", string(m.diff),
+		format.Message(other, "expected to match", m.original))
+}
+
+// NegatedFailureMessage returns a string comparing the full objects after an unexpected match has occurred.
+func (m *Matcher) NegatedFailureMessage(other interface{}) (message string) {
+	return format.Message(other, "not to match", m.original)
+}
+
+// calculateDiff applies the MatchOptions and identified the diff between the Matcher object and the passed object.
```

Comment:
```suggestion
// calculateDiff applies the MatchOptions and identifies the diff between the Matcher object and the passed object.
```
nit, (or maybe calculates)